chalk-integration/src/lowering.rs

```diff
@@ -86,7 +86,8 @@ struct AssociatedTyLookup {
     addl_variable_kinds: Vec<chalk_ir::VariableKind<ChalkIr>>,
 }
 
-enum ApplyTypeLookup {
+enum ApplyTypeLookup<'k> {
+    Param(&'k chalk_ir::WithKind<ChalkIr, BoundVar>),
```

Comment:
Hmm, so. I'm not super sure about this name, because I don't want it to be confused with the rustc `Param` type.

Reply:
Hmm, not sure what else to go with, since `BoundVar` would be confusing as well.

Reply:
I guess it ends up serving a similar purpose to rustc's `Param`, so if we can't think of a better name, `Param` is probably acceptable. Another option: `Parameter`. Feel free to take it or leave it.